jerry-core/ecma/builtin-objects/ecma-builtins.c

```diff
@@ -949,6 +949,40 @@ ecma_builtin_dispatch_routine (ecma_builtin_id_t builtin_object_id, /**< built-i
                                ecma_length_t arguments_list_len) /**< length of arguments' list */
 {
   JERRY_ASSERT (builtin_object_id < ECMA_BUILTIN_ID__COUNT);
+
+  if (arguments_list_len <= 3)
+  {
+    ecma_value_t padded_arguments_list_p[3] = { ECMA_VALUE_UNDEFINED, ECMA_VALUE_UNDEFINED, ECMA_VALUE_UNDEFINED };
+
+    switch (arguments_list_len)
+    {
+      case 3:
+      {
+        padded_arguments_list_p[2] = arguments_list_p[2];
+        /* FALLTHRU */
+      }
+      case 2:
+      {
+        padded_arguments_list_p[1] = arguments_list_p[1];
+        /* FALLTHRU */
+      }
+      case 1:
+      {
+        padded_arguments_list_p[0] = arguments_list_p[0];
+        break;
+      }
+      default:
+      {
+        JERRY_ASSERT (arguments_list_len == 0);
+      }
+    }
```

Comment:
`arguments_list_p = padded_arguments_list_p` isn't missing from here?

Reply:
I think is not missing. `padded_arguments_list_p` is a local array inside the `if (arguments_list_len <= 3)` condition.

Reply:
But if you do that you don't need two `ecma_builtin_routines[builtin_object_id] (...)` calls, only one at the end of the function.

Reply:
Okay but for this case the `padded_arguments_list_p` must be in the function scope not in the `if ()`'s scope since it is a local array. I am not insist to the current solution, @akosthekiss what your opinion?

Reply:
From technical perspective: it is always at func scope because messing with function stack allocation is a costly which compilers try to avoid. If you really want special casing this move to a never inline helper function, which can be called by a tail call, or create two functions.

Reply:
Okay, you convinced me!